cmd/main.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	cobbler "github.com/ContainerSolutions/cobblerclient"
+)
+
+var config = cobbler.ClientConfig{
+	Url:      "http://localhost:25151",
+	Username: "cobbler",
+	Password: "password",
+}
+
+func main() {
+	c := cobbler.NewClient(http.DefaultClient, config)
+	_, err := c.Login()
+	if err != nil {
+		fmt.Printf("Error logging in: %s\n", err)
+	}
+
+	fmt.Printf("Token: %s\n", c.Token)
+
+	d := cobbler.Distro{
+		Name:      "Test",
+		Breed:     "Ubuntu",
+		OSVersion: "trusty",
+		Arch:      "x86_64",
+		Kernel:    "/var/www/cobbler/ks_mirror/Ubuntu-14.04/install/netboot/ubuntu-installer/amd64/linux",
+		Initrd:    "/var/www/cobbler/ks_mirror/Ubuntu-14.04/install/netboot/ubuntu-installer/amd64/initrd.gz",
+	}
+
+	fmt.Println("Creating a Distro")
+	newDistro, err := c.CreateDistro(d)
+	if err != nil {
+		fmt.Printf("Error creating distro: %s\n", err)
+	}
+
+	fmt.Printf("New Distro: %+v\n", newDistro)
+
+	if newDistro.Name != "Test" {
+		fmt.Println("Distro name does not match.")
+	}
+
+	fmt.Println("Updating Distro")
+	newDistro.Comment = "Update Test"
+	if err := c.UpdateDistro(newDistro); err != nil {
+		fmt.Printf("Error creating distro: %s\n", err)
+	}
+
+	fmt.Println("Creating a System")
+	eth0 := cobbler.Interface{
+		MACAddress:    "aa:bb:cc:dd:ee:ff",
+		Static:        true,
+		InterfaceType: "bridge",
+	}
+
+	eth1 := cobbler.Interface{
+		MACAddress: "aa:bb:cc:dd:ee:fa",
+		Static:     true,
+	}
+
+	s := cobbler.System{
+		Comment:     "WTF",
+		Name:        "Foobar",
+		Profile:     "Ubuntu-14.04-x86_64",
+		NameServers: []string{"8.8.8.8", "1.1.1.1"},
+		PowerID:     "foo",
+	}
+
+	newSystem, err := c.CreateSystem(s)
+	if err != nil {
+		fmt.Printf("Error creating system: %s\n", err)
+	}
+
+	fmt.Printf("New System: %+v\n", newSystem)
+
+	fmt.Println("Adding NIC to System")
```

Comment:
This line and the rest below it that reference to `newSystem` will fail if the system is not created.